pkg/providers/pricing/suite_test.go

```diff
@@ -262,6 +262,44 @@ var _ = Describe("Pricing", func() {
 		Expect(lo.Map(inp.ProductDescriptions, func(x *string, _ int) string { return *x })).
 			To(ContainElements("Linux/UNIX", "Linux/UNIX (Amazon VPC)"))
 	})
+	It("should return static on-demand data when in isolated-vpc", func() {
+		ctx = options.ToContext(ctx, test.Options(test.OptionsFields{
+			IsolatedVPC: lo.ToPtr(true),
+		}))
+		now := time.Now()
+		awsEnv.EC2API.DescribeSpotPriceHistoryOutput.Set(&ec2.DescribeSpotPriceHistoryOutput{
+			SpotPriceHistory: []*ec2.SpotPrice{
+				{
+					AvailabilityZone: aws.String("test-zone-1b"),
+					InstanceType:     aws.String("c99.large"),
+					SpotPrice:        aws.String("1.50"),
+					Timestamp:        &now,
+				},
+				{
+					AvailabilityZone: aws.String("test-zone-1b"),
+					InstanceType:     aws.String("c98.large"),
+					SpotPrice:        aws.String("1.10"),
+					Timestamp:        &now,
+				},
+			},
+		})
+
+		awsEnv.PricingAPI.GetProductsOutput.Set(&awspricing.GetProductsOutput{
+			PriceList: []aws.JSONValue{
+				fake.NewOnDemandPrice("c98.large", 1.20),
+				fake.NewOnDemandPrice("c99.large", 1.23),
+			},
+		})
+		ExpectReconcileSucceeded(ctx, controller, types.NamespacedName{})
+		price, ok := awsEnv.PricingProvider.OnDemandPrice("c5.large")
```

Comment:
nit: Should we get an on-demand price for the `c98.large` since we have a price at our API for it, but we want to get the static price. This should give us the confidence that we really aren't hitting our API, since we aren't using the value that we explicitly set in the test-case.

Reply:
Done